a-trivial-procore-react-component.md

```diff
@@ -1,3 +1,171 @@
 # A Trivial React Component
 
-> by: &lt;YOUR-NAME-HERE&gt;
+> by: Michael Cola
+
+# React Components in `procore/procore` and `procore/wrench`
+
+### Why use `react_component`?
+* The problem may have been solved by someone else, you don't need to reinvent the wheel.
+* Back End engineers need only produce the properties necessary to implement a shared component, and don't have to worry about creating the component in wrench, only where it is on the page
+
+## Usage
+
+### Requirements
+* Component is exposed to the `Window` in `procore/wrench` via [`src/components.js`](https://github.com/procore/wrench/blob/master/src/components.js)
+	* Component must be defined and exported from the file location defined in `src/components.js`
+* Usage in procore is on an `.erb` template, surrounding by ruby HTML interpolation: `<%= react_component() %>`
+
+### API for the `react_component` method
+* in [`app/helpers/application_helper.rb`](https://github.com/procore/procore/blob/master/app/helpers/application_helper.rb)
+* `react_component` takes in a component class name, and the props necessary for that component to render, as a hash
+  * Note: the keys in the hash are the same keys being read in wrench for the props on the component
+	* Ex:
+```ruby
+react_component(
+  "ExampleComponent",
+  {
+    links: @links
+  }
+)
+```
+### Component exposition in wrench
+* Add a line to `src/components.js`
+	* `import ExampleComponent from 'components/ExampleComponent';`
+	* Component definition:
+```jsx
+import React, { PropTypes } from 'react';
+
+class ExampleComponent extends React.Component {
+  static propTypes = {
+    editable: PropTypes.bool,
+    links: PropTypes.object.isRequired,
+  };
+
+  static defaultProps = {
+    editable: false,
+  }
+
+  render() {
+    return (
+      <a href={links.redirect} >
+        Take me to your leader
+      </a>
+    );
+  }
+}
+export default ExampleComponent;
+```
+**Note:** You may notice a different syntax for defining a component in wrench
+```jsx
+import { PropTypes } from 'react';
+
+const ExampleComponent = (props) => {
+  const { links } = props;
+
+  return (
+    <a href={links.redirect} >
+      Take me to your leader
+    </a>
+  )
+};
+
+ExampleComponent.propTypes = {
+  links: PropTypes.object.isRequired,
+  editable: PropTypes.bool,
+};
+
+ExampleComponent.defaultProps = {
+  editable: false,
+};
+
+export default ExampleComponent;
+```
+This is called a "Pure Component".
+
```

Comment:
Only thing I'd add here is an example of how `import` or `System.import`ing this component looks like in `src/components.js` and how it looks like when we add it to the `export default` block.